ansible/playbooks/aix.yml

```diff
@@ -6,125 +6,282 @@
-      file: path=/tmp/playbook_tmp/xlc state=directory mode=0755
-      tags: layout
-
-    ################################
-    # AIX filesystem configuration #
-    ################################
+  ################################
+  # AIX filesystem configuration #
+  ################################
     - name: Download AIX filesystem configuration script
       get_url:
         url: https://raw.githubusercontent.com/AdoptOpenJDK/openjdk-infrastructure/master/ansible/playbooks/scripts/AIX_filesystem_config.sh
-        dest: /tmp/playbook_tmp/AIX_filesystem_config.sh
+        dest: /tmp/AIX_filesystem_config.sh
         mode: 0775
         validate_certs: no
       tags: filesystem
         
     - name: Execute AIX filesystem configuration script
-      command: /tmp/playbook_tmp/AIX_filesystem_config.sh
+      command: /tmp/AIX_filesystem_config.sh
       tags: filesystem
 
-    ######################################
-    # Install and configure RPM Packages #
-    ######################################
-    - name: Transfer RPM Packages to remote host
+    - file:
+        state: absent
+        path: /tmp/AIX_filesystem_config.sh
+      tags: filesystem
+
+  ##########################################################################
+  # Install openssl from IBM installp                                      #
+  # openssl installp download requiring an IBMid                           #
+  # https://www-01.ibm.com/support/docviepw.wss?uid=isg1fileset-1190419011 #
+  ##########################################################################
+    - name: Transfer Openssl to remote host
       copy:
-        src: "{{ item }}"
-        dest: /tmp/playbook_tmp/rpms/
-        mode: 0755
-      with_fileglob:
-        - /home/ubuntu/aix/rpms/*
-      tags: additional_rpms
+        src: /Vendor_Files/aix/openssl-1.0.2.1300.tar.gz
+        dest: /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+
+    - name: Extract Openssl 
+      shell: gzip -d /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+    
+    - shell: tar -xvf /tmp/openssl-1.0.2.1300.tar -C /tmp/
+      tags: openssl
+
+    - name: Install IBM Openssl - installp
+      shell: installp -aXYgd /tmp/openssl-1.0.2.1300 openssl.base
+      register: result.openssl
+      ignore_errors: yes
+      tags: openssl
 
-    - name: Download script to install RPM packages
+    - name: Clean openssl tmp files
+      file:
+        path: "{{ item }}"
+        state: absent
+      with_items:
+        - /tmp/openssl-1.0.2.1300
+        - /tmp/openssl-1.0.2.1300.tar
+      tags: openssl
+
+  ####################################################
+  # Uninstall conflicting packages from base image   #
+  # if they were installed via rpm unless yum exists #
+  ####################################################
+    - stat:
+        path: /usr/bin/yum
+      register: yum
+
+    - debug:
+        msg: "Yum installed, skipping download and installation"
+      when: yum.stat.islnk is defined
```

Comment:
``` when: yum.stat.islnk is defined ```

Is ``` yum ``` a symlink?
```

islnk | Tells you if the path is a symbolic link | success, path exists and user can read stats | boolean | False

```

Should it be ```yum.stat.exists == True ```

Reply:
Technically, stat is on /usr/bin/yum which is a link to /opt/freeware/bin/yum, both of which are added with the yum.sh 'install'.

I think the method is valid, but I could also change to exists method if you feel strongly about it.

Reply:
no, that's fine just wanted to confirm